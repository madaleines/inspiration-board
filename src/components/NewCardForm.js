import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"];

class NewCardForm extends Component {
  constructor(props) {
    super(props);

    const resetState = {
      text: '',
      emoji: ''
    };

    this.state = {...resetState};
    this.resetState = {...resetState}
  }

  onFieldChange = (event) => {
    const { name, value } = event.target;
    const updatedField = {};
    updatedField[name] = value;
    this.setState(updatedField);
  };

  submitForm = (event) => {
    event.preventDefault();
    const newCard = {...this.state};
    this.props.addCardCallback(newCard);
    this.setState(this.resetState);
  };

  getEmojiList = () => {
  const options = EMOJI_LIST.map((text, index) => {
    const emojiUnicode = emoji.getUnicode(text);
    return <option key={ index } value={ text }>{ emojiUnicode }</option>
  });
  return options;
};

  render() {
    return (
      <section className="new-card-form">
        <h3 className="new-card-form__header">Create a New Inspiration</h3>
        <form onSubmit={ this.submitForm } className="new-card-form__form">
          <label htmlFor="text" className="new-card-form__form-label">Message: </label>
          <textarea name="text" value={this.state.text} onChange={ this.onFieldChange } className="new-card-form__form-textarea"/>
          <label htmlFor="emoji" className="new-card-form__form-label">Emoji: </label>
          <select name="emoji" value={this.state.emoji} onChange={ this.onFieldChange } className="new-card-form__form-select">
            { this.getEmojiList() }
          </select>
          <input type="submit" value="create card" className="new-card-form__form-button"/>
        </form>
      </section>
    )
  }
}

NewCardForm.propTypes = {
  addCardCallback: PropTypes.func.isRequired
};

export default NewCardForm;
