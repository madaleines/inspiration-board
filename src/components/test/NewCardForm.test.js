import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import NewCardForm from '../NewCardForm';

describe('NewCardForm', () => {

  test('that it renders NewCardForm with shallow rendering', () => {
    const wrapper = shallow(<NewCardForm submitNewCard={ ()=>{ } } />);

    expect(wrapper).toMatchSnapshot();

    wrapper.unmount();
  });

});
