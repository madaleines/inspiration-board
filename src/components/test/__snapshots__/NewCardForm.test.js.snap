// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it renders NewCardForm with shallow rendering 1`] = `
<section
  className="new-card-form"
>
  <h3
    className="new-card-form__header"
  >
    Create a New Inspiration
  </h3>
  <form
    className="new-card-form__form"
    onSubmit={[Function]}
  >
    <label
      className="new-card-form__form-label"
      htmlFor="text"
    >
      Message: 
    </label>
    <textarea
      className="new-card-form__form-textarea"
      name="text"
      onChange={[Function]}
      value=""
    />
    <label
      className="new-card-form__form-label"
      htmlFor="emoji"
    >
      Emoji: 
    </label>
    <select
      className="new-card-form__form-select"
      name="emoji"
      onChange={[Function]}
      value=""
    >
      <option
        key="0"
        value=""
      />
      <option
        key="1"
        value="heart_eyes"
      >
        😍
      </option>
      <option
        key="2"
        value="beer"
      >
        🍺
      </option>
      <option
        key="3"
        value="clap"
      >
        👏
      </option>
      <option
        key="4"
        value="sparkling_heart"
      >
        💖
      </option>
      <option
        key="5"
        value="heart_eyes_cat"
      >
        😻
      </option>
      <option
        key="6"
        value="dog"
      >
        🐶
      </option>
    </select>
    <input
      className="new-card-form__form-button"
      type="submit"
      value="create card"
    />
  </form>
</section>
`;
